cryptography/src/commonMain/kotlin/com/wire/kalium/cryptography/utils/CryptoUtils.kt

```diff
@@ -1,3 +1,7 @@
 package com.wire.kalium.cryptography.utils
 
 expect fun calcMd5(bytes: ByteArray): String
+
+expect fun encryptDataWithAES256(data: ByteArray): Pair<ByteArray,ByteArray>
```

Comment:
I don't have a preference, but this is also an option:

```kotlin
@JvmInline
value class AES256Key(val data: ByteArray)

@JvmInline
value class EncryptedData(val data: ByteArray)

@JvmInline
value class PlainData(val data: ByteArray)
```

So we prevent consumers of `cryptography` from mistakenly calling functions with wrong parameters, as there are a lot of `ByteArray`s.

Usage:

```kotlin
val plainData = PlainData(assetData)
val key = generateRandomAES256Key()
val encryptedData: EncryptedData = encryptDataWithAES256(plainData, key)
```

Reply:
I added a `SymmetricSecretKey` typealias instead. Let me know @typfel if that is enough for you or if you would rather I take the suggested change from @vitorhugods

Reply:
If you don't feel strongly about it my preference would be to go for this suggestion from @vitorhugods:
```
expect fun generateRandomAES256Key(): AES256Key

expect fun encryptDataWithAES256(data: ByteArray, key: AES256Key = generateRandomAES256Key()): ByteArray
```

Why? Because It's a more flexible API, maybe we have use case in the future were we need generate the key before we have the data we want to encrypt.

---

If we introduce a type alias like `SymmetricSecretKey` I think we should name it `AES256Key` since the key is bound to the AES256 encryption.

Reply:
Added Vitor's suggestion to encapsulate both byteArray into more comprehensive objects

Reply:
Nice!